src/SFA.DAS.ApprenticeAccounts.Web/Pages/Account.cshtml.cs

```diff
@@ -39,9 +40,11 @@ public AccountModel(ApprenticeApi api, NavigationUrlHelper urlHelper)
         public bool IsCreating { get; private set; } = false;
 
         public async Task<IActionResult> OnGetAsync(
-            [FromServices] AuthenticatedUser user)
+            [FromServices] AuthenticatedUser user,
+            [FromQuery] string returnUrl)
         {
             ViewData.SetWelcomeText("Welcome");
+            TempData[ReturnUrlKey] = returnUrl;
```

Comment:
What happens when the returnUrl is blank or an empty string or null. Will it still be saved to the TempData?

Reply:
Yes, I need the temp data set anyway so I can make decision on post.

Reply:
Well if the RedirectURL ever ends up being an empty string. we will redirect he user to a blank page (ie if TempData.Get returns true, we will end up redirecting).  


Looking at it a bit closer. The postback is on the same page, so we don't have to use TempData. The returnUrl is still in the URL when we do OnPost(), So we can just check it on PostBack and redirect accordingly. 

I believe the general consensus is not to use TempData if we don't need to

Reply:
@Pauljgraham This is fixed.